ipwb/__main__.py

```diff
@@ -44,10 +44,10 @@ def checkArgs_replay(args):
         random.seed()
         # Write data to temp file (sub-optimal)
 
-        tf = tempfile.NamedTemporaryFile(mode='w', suffix='.cdxj')
-        tf.write(cdxjIn)
-        args.index = tf.name
-        tf.close()
+        fh, args.index = tempfile.mkstemp(suffix='.cdxj')
+        with open(fh, 'w') as f:
+            f.write(cdxjIn)
```

Comment:
They suggests using `os.close(fh)` to properly close files created like this. I would suggest you immediately close the `fh` after creating it, then using your `with open` block on the `args.index`. Also, a similar code is present in the testUtil file, that needs to be fixed as well.

Reply:
> They

who?

Doesn't only opening it in the `with` block close it when the scope is escaped?

> using your with open block on the args.index

What does this mean?

Reply:
> who?

All those smart people on the web.

> Doesn't only opening it in the `with` block close it when the scope is escaped?

Yes, it does, but I am afraid we are attempting to open an existing file handle, not a file name. Opening a file handle works, but I am cautious here because many people suggested that file handle be closed using `os.close(fh)` and not the usual `fh.close()`, I did not investigate why do they say that though.

> What does this mean?

I meant something like this:

```
        fh, args.index = tempfile.mkstemp(suffix='.cdxj')
        os.close(fh)
        with open(args.index, 'w') as f:
            f.write(cdxjIn)
```

Don't forget to import `os` package, if not imported already.